lib/zstd.h

```diff
@@ -495,23 +495,38 @@ ZSTDLIB_API size_t ZSTD_sizeof_DDict(const ZSTD_DDict* ddict);
 
 /*! ZSTD_estimate*() :
  *  These functions make it possible to estimate memory usage
- *  of a future target object, before its allocation,
- *  given a set of parameters, which vary depending on target object.
+ *  of a future {D,C}Ctx, before its creation.
  *  The objective is to guide decision before allocation.
+ *  ZSTD_estimateCCtxSize() will consider src size to be arbitrarily "large".
+ *  If srcSize is known to be small, ZSTD_estimateCCtxSize_advanced() can provide a tighter estimation.
+ *  ZSTD_estimateCCtxSize_advanced() can be used in tandem with ZSTD_getCParams() to create cParams from compressionLevel.
  *  Note : CCtx estimation is only correct for single-threaded compression */
-ZSTDLIB_API size_t ZSTD_estimateCCtxSize(ZSTD_compressionParameters cParams);
+ZSTDLIB_API size_t ZSTD_estimateCCtxSize(int compressionLevel);
```

Comment:
OK, I've tested a modified compression level table to guarantee that memory budget is monotonically increasing (I found a few more cases besides level 4->5).
The new compression level table still guarantees that higher compression levels compress more, at least on `silesia.tar`, and that lower compression levels compress faster, at least on target host (Chip's Entropy).
What is slightly impacted is the regularity of the speed slowdown. And still, the impact is quite moderate.

As a plus side, I believe the regularity of speed decrease was really too specific to target host, since `paramgrill` calculated it on this machine, and it would be different on another host (especially with different cache size). I actually expect the monotonically increase in memory budget to smooth things for _other_ hosts. So I'm inclined to adopt the new table.

In itself, it's a good enough quick fix for `ZSTD_estimateCCtxSize()`, where this property _feels implied_.

Now, thing is, a future change _might_ impact this property.
So it could be documented.
It could also be enforced within `paramgrill` so that it respects this property while generating a new compression levels table.

And lastly, `ZSTD_estimateCCtxSize()` could internally implement a "max of all lower levels memory usage" policy, even if it's not useful with the new compression level table, just to be on the safe side regarding future changes.